h/services/group.py

```diff
@@ -88,21 +89,31 @@ def groupids_readable_by(self, user, group_ids=None):
 
         :type user: `h.models.user.User`
         """
-        readable = Group.readable_by == ReadableBy.world
+        # Very few groups are readable by world, query those separately
+        readable_by_world_query = select(Group.pubid).where(
+            Group.readable_by == ReadableBy.world
+        )
+        query = readable_by_world_query
 
         if user is not None:
-            readable_member = sa.and_(
-                Group.readable_by == ReadableBy.members,
-                Group.members.any(User.id == user.id),
```

Comment:
Note that the members relationship joins with both GroupMembership and User.

We don't care about any column in user, we can filter directly on the membership talble.